tests/TestCase/Database/Driver/MysqlTest.php

```diff
@@ -163,4 +163,138 @@ public function testCommitTransactionAutoConnect()
         $this->assertFalse($driver->commitTransaction());
         $this->assertTrue($driver->isConnected());
     }
+
+    /**
+     * Test for utf8mb4 Failure #1
+     *
+     * @expectedException \PDOException
+     * @expectedExceptionMessage SQLSTATE[HY000]: General error: 1366 Incorrect string value: '\xF0\x9F\x98\x83
+     * @return void
+     */
+    public function testUtf8mb4FailureOnConnectionMode()
+    {
+        $sqlMode = 'SET SESSION sql_mode = "STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION";';
+        $dropStm = "DROP TABLE IF EXISTS `test_utf8mb4`;";
+        $createStm = "CREATE TABLE `test_utf8mb4` (`field` TEXT CHARACTER SET utf8mb4 NOT NULL) ENGINE=INNODB;";
+        $insertStm = "INSERT INTO test_utf8mb4 SET field = '😃';";
+
+        $connection = new \Cake\Database\Connection([
+            'driver' => '\Cake\Database\Driver\Mysql',
+            'persistent' => true,
+            'host' => 'localhost',
+            'username' => 'root',
+            'password' => '',
+            'database' => 'cakephp_test',
+            'port' => '3306',
+            // Try to insert utf8 4-byte symbol through utf8 3-byte connection character set
```

Comment:
What is this test testing? It looks like it is testing MySQL, I'm not sure we need to do that. The tests above ensure that the DSN sent to PDO is correctly built which is enough for me.

Reply:
Then I assume correctly that none of the new tests are required for this PR to become mergable, correct?

Reply:
I don't think this test is necessary at least.

Reply:
Well the DSN is tested above by another unit. I will remove all the other tests.